### PR TITLE
Added Heartbeat Actor (#51)

### PR DESCRIPTION
**Pull request by rishflab**
Implemented an actor that emits a heartbeat periodically. This periodic
heartbeat functionality can be used to send sync requests to other nodes.

This commit demonstrates:
1. how to run actix actors in conjunction with actix-web
2. how to pass messages to an actor to mutate/query its state

### CODE REVIEW

File: lachesis-rs/src/server/handlers.rs [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        .from_err()
+        .and_then(|val: SubmitTransaction| {
+            println!("model: {:?}", val);
+            Ok(HttpResponse::Ok().json(val)) // <- send response
+        })
+        .responder()
+}
+
+pub fn heartbeat(req: &HttpRequest<AppState>) -> HttpResponse {
+    println!("{:?}", req);
```
Comment (Maxime2) [inline lachesis-rs/src/server/handlers.rs]: Do we need this line in production, or it is a debugging leftover?

**Comment by Maxime2** [comment]
We need an exclusive heartbeat functionality which guarantees no new sync request initiated until previous one finishes, even heartbeat interval has passed since previous sync started. That would guarantee we will not overload a node with many concurrent syncs.

File: lachesis-rs/src/server/handlers.rs (hunk 1 above)
Comment (rishflab) [reply lachesis-rs/src/server/handlers.rs]: debugging left over, needs to be removed

**Comment by rishflab** [comment]
@Maxime2 yes, the sync request dispatch needs to be in an actor, in a loop with a configurable delay after each dispatch.

**Pull request merged**